pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/rebalance/TableRebalancer.java

```diff
@@ -598,9 +611,12 @@ private RebalanceResult doRebalance(TableConfig tableConfig, RebalanceConfig reb
             "Rebalance has stopped already before updating the IdealState", instancePartitionsMap,
             tierToInstancePartitionsMap, targetAssignment, preChecksResult, summaryResult);
       }
+      String partitionColumn = TableConfigUtils.getPartitionColumn(tableConfig);
       Map<String, Map<String, String>> nextAssignment =
           getNextAssignment(currentAssignment, targetAssignment, minAvailableReplicas, enableStrictReplicaGroup,
```

Comment:
Seems most of the new arguments are used to calculate the partition id, consider passing a function (lambda) to simplify this

Reply:
done